### ISSUE TITLE
Marty.renderToString and nested routes.

### ISSUE THREAD

**Issue by ntkoso** [martyjs/marty#314]
Routes:
```javascript
<Route name='Root' path='/' handler={App}>
  <Route name='SomePage' handler={SomePage}>
    <Route name='SomeSection' handler={SomeSection}/>
  </Route>
</Route>
```
`App` - wrapped in container with fetches(eg. asset paths). renderToString waits on fetches and returns proper html and diagnostics.
`SomePage` - wrapped in container with fetches(eg. some dynamic menu elements and sidebar data). renderToStrings starts fetches, but returns only App fetches, without waiting on `SomePage` fetches.
`SomeSection` - wrapped in container with fetches(eg. user lists and other data). Same problem here.

Nested fetches are not timing out, they are completed under 100ms. 
They are not registered in diagnostics. 
They are not waited on, but they are executed outside of renderToString. renderToString starts them, but returns html as soon as `App` fetches are ready.
Server already sent finalized html, but nested fetches are still been executed in background.
When container is removed from `App`, `SomePage` fetches are waited on and added to diagnostics.

Same code is working on client side without problems.

**Comment by jhollingworth** [comment]
I think I've worked out what the issue is.

In Marty.renderToString we

1. Render a component tree
2. Check to see if any components made fetches and wait for them
3. Re-render the component tree

This makes the assumption that your components will not make subsequent fetches as the result of the initial fetch. This assumption falls down because the page container won't render the section component until it has finished all of its fetches.

I will update renderToString to continue trying to re-render until there are no more fetches.

**Comment by jhollingworth** [comment]
Well that was a fun bug! I've just released v0.9.15 which resolves this issue
